pkg/config/config.go

```diff
@@ -322,7 +319,8 @@ func (dc *DeploymentConfig) listUnusedModules() map[string][]string {
 	unusedModules := make(map[string][]string)
 	for _, conn := range dc.moduleConnections {
 		if conn.isEmpty() {
-			unusedModules[conn.fromID] = append(unusedModules[conn.fromID], conn.toID)
+			fromMod := conn.ref.getFromModuleID()
+			unusedModules[fromMod] = append(unusedModules[fromMod], conn.ref.getToModuleID())
```

Comment:
nit: OCD doesn't like the different treatment of from and to here. One is stored separate and the other is inline. You could initialize both on one line (L322) with something like `x, y := a, b`.

Reply:
My thinking here was to avoid 2 func calls rather than cleanliness. If you bump this above nit, glad to address it. Otherwise, "ack".